### PR TITLE
Convert first pages to typescript

### LINKED ISSUE #3729: Consider changing copy on "Unauthorized" page
Comment from @hatched in relation to being shown an unauthorized page when trying to access an admin section as a viewer:

> Would the user ever get this unexpectedly? If so maybe adding something about "Contact x if you think you're getting this message in error".

### LINKED ISSUE #3768: Write tests for brand store NoStores view

### LINKED ISSUE #3769: Convert brand store NoStores view to TypeScript

### LINKED ISSUE #3770: Convert brand store NotAuthorized view to TypeScript

### LINKED ISSUE #3771: Write tests for brand store NotAuthorized view

### PR DESCRIPTION
**Pull request by steverydz**
## Done
Added tests and converted `NoStores` and `NotAuthorized` to TypeScript

## QA
- Check that the code builds
- Check that the tests pass, e.g. `dotrun exec yarn test-js static/js/brandstore`

## Issues
Fixes #3771
Fixes #3768
Fixes #3769
Fixes #3770
Fixes #3729

### CODE REVIEW

File: static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import NotAuthorized from "./NotAuthorized";
+
+test("the 'is public' checkbox should not be checked when the current store is set to private", () => {
```
Comment (hatched) [inline static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx:6]: There isn't actually a checkbox in this component....is this a copy/paste error?

File: tsconfig.json [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+    "noFallthroughCasesInSwitch": true,
     "moduleResolution": "node",
-    "strict": true
-  }
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "noEmit": false,
+    "sourceMap": true
+  },
+  "include": ["./static/js/brand-store/NoStores"]
```
Comment (hatched) [inline tsconfig.json:22]: You may now be able to add just `./static/js` and have it work without specifying each folder manually.

File: static/js/brand-store/NoStores/NoStores.test.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import NoStores from "./NoStores";
+
+test("the 'is public' checkbox should not be checked when the current store is set to private", () => {
```
Comment (hatched) [inline static/js/brand-store/NoStores/NoStores.test.tsx:6]: There is no checkbox in this component?

**Review by hatched — CHANGES_REQUESTED** [review]

File: static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx (hunk 1 above)
Comment (steverydz) [reply static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx]: sorry - doing too many things at once! fixed

File: static/js/brand-store/NoStores/NoStores.test.tsx (hunk 3 above)
Comment (steverydz) [reply static/js/brand-store/NoStores/NoStores.test.tsx]: sorry - doing too many things at once! fixed

File: tsconfig.json (hunk 2 above)
Comment (steverydz) [reply tsconfig.json]: Yes, it does! Thanks, updated

File: static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import NotAuthorized from "./NotAuthorized";
+
+test("the page should have the correct title", () => {
+  render(<NotAuthorized />);
+
+  expect(screen.getByText("Not authorized")).toBeInTheDocument();
```
Comment (petermakowski) [inline static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx:9]: You could check for heading (even specifying level 1) with getByRole to be more explicit. It is clearly stated in the test description we expect a title.

**Review by hatched — APPROVED** [review]
Thanks for the continued work on this 👍

File: static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx (hunk 4 above)
Comment (steverydz) [reply static/js/brand-store/NotAuthorized/NotAuthorized.test.tsx]: Thanks - done

**Pull request merged**